src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -266,13 +266,62 @@ private static double computeTigExplainQualOfOneAlignment(final List<AlignmentIn
         return numFirst.thenComparing(mismatchSecond);
     }
 
-    private static List<AlignmentInterval> splitGaps(final List<AlignmentInterval> configuration) {
-        return configuration.stream()
+    @VisibleForTesting
+    static List<AlignmentInterval> splitGaps(final List<AlignmentInterval> configuration) {
+
+        final List<Integer> blackList = new ArrayList<>(configuration.size());
+
+        for (int i = 0; i < configuration.size(); ++i) {
+            final AlignmentInterval alignment = configuration.get(i);
+            if ( alignmentContainsLargeGap(alignment) ) {
+                final Iterable<AlignmentInterval> gapSplitAlignments =
+                        ContigAlignmentsModifier.splitGappedAlignment(alignment, GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY,
+                                SvCigarUtils.getUnclippedReadLength(alignment.cigarAlong5to3DirectionOfContig));
+
+                for (int j = 0; j < configuration.size(); ++j) {
+                    final AlignmentInterval other = configuration.get(j);
+                    if (j == i || AlignmentInterval.overlapOnContig(alignment, other) == 0)
+                        continue;
+
+                    if ( Utils.stream(gapSplitAlignments).anyMatch(other::containsOnRead) ) {
+                        if ( gappedAlignmentOffersBetterCoverage(alignment, other) ) {
+                            blackList.add(j);
+                        } else {
+                            blackList.add(i);
+                        }
+                    }
+                }
+            }
+        }
+
+        final List<AlignmentInterval> result = new ArrayList<>(configuration.size());
+        for (int i = 0; i < configuration.size(); ++i) {
+            if (blackList.indexOf(i) == -1)
```

Comment:
I think `contains(i)` would be easier to read.

Reply:
irrelevant now as switched to the suggested logic